omnibus/files/server-ctl-cookbooks/infra-server/libraries/preflight_indexing_validator.rb

```diff
@@ -130,6 +131,23 @@ def verify_erchef_config
     end
   end
 
+  def verify_search_engine_permissions
+    if node['private_chef']['opscode-erchef']['search_provider'] != 'solr'
+      search_provider_url = helper.solr_url
+      begin
+        client = Chef::HTTP.new("#{search_provider_url}/_all/_settings")
+        response = JSON.parse(client.get(''))
+        if response['chef']['settings']['index'].key?('blocks')
```

Comment:
These can be combined with `response.dig('chef', 'settings', 'index', 'blocks', 'read_only_allow_delete)`